src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncTests.cs

```diff
@@ -31,6 +31,159 @@ private CompilationVerifier CompileAndVerify(string source, string expectedOutpu
         }
 
         [Fact]
+        public void AsyncEmitMainOfIntTest()
```

Comment:
Perhaps extract `async Main` tests to a separate `CodeGenAsyncMainTests.cs` file. That would group the tests and would allow tests to have simpler names.